tests/tests_core/test_early_stopping.py

```diff
@@ -7,23 +7,20 @@
 
 
 class EarlyStoppingTestCase(unittest.TestCase):
-    """
-      Defines the setUp() and tearDown() functions for the early-stopping test cases.
-      """
+    """ Defines the setUp() and tearDown() functions for the early-stopping test cases."""
 
     def setUp(self):
-        """
-        Load the needed datasets for all EarlyStoppingTestCases and set the random
+        """ Load the needed datasets for all EarlyStoppingTestCases and set the random
         seed so that randomized algorithms show deterministic behaviour.
         """
         np.random.seed(0)
 
-        self.rand_s1 = np.random.normal (loc=0.0, size=1000)
-        self.rand_s2 = np.random.normal (loc=0.1, size=1000)
-        self.rand_s3 = np.random.poisson(lam=1.0, size=1000)
-        self.rand_s4 = np.random.poisson(lam=3.0, size=1000)
-        self.rand_s5 = np.random.normal (loc=0.0, size=1000)
-        self.rand_s6 = np.random.normal (loc=0.1, size=1000)
+        self.rand_s1 = list(np.random.normal (loc=0.0, size=1000))
+        self.rand_s2 = list(np.random.normal (loc=0.1, size=1000))
+        self.rand_s3 = list(np.random.poisson(lam=1.0, size=1000))
+        self.rand_s4 = list(np.random.poisson(lam=3.0, size=1000))
+        self.rand_s5 = list(np.random.normal (loc=0.0, size=1000))
+        self.rand_s6 = list(np.random.normal (loc=0.1, size=1000))
```

Comment:
after the updated check for ```ndarray```, maybe we don't need the ```list()``` function here.

Reply:
then all tests that test statistics will fail. I can create additional test to test the correct input type.